src/storage/src/hummock/iterator/forward_user.rs

```diff
@@ -102,15 +103,14 @@ impl<I: HummockIterator<Direction = Forward>> UserIterator<I> {
             }
 
             if &self.last_key.user_key.as_ref() != key {
-                self.last_key.set(full_key);
+                self.last_key = full_key.to_vec().into_bytes();
```

Comment:
need a new function `to_bytes` to avoid extra one `to_vec`

Reply:
As we support more table key types for `FullKey`, we will need a lot of conversion functions. Perhaps the conversion of `FullKey<T>` be consistent with the conversion of `T`? For example, to convert a `&[u8]` to `Bytes`, we also need to convert it to `Vec` first, then to `Bytes`. Supporting direct conversion of every `T` will probably clutter up the code.

Reply:
I used to try `impl<'a, T: From<&'a [u8]>> From<FullKey<&'a [u8]>> for FullKey<T>` but this implementation conflicts with `impl<T> std::convert::From<T> for T` because `&'a [u8]` itself also `impl From<&'a [u8]>`.

This seems to be a known limitation of rust and have been complained for long, but no solution has come up.

Reply:
I have implemented a generic method named `copy_into` that convert the `FullKey<&[u8]>` into `FullKey<T>` for T that can be creating by copying from a slice (a new trait named `CopyFromSlice`).